algorithms.tex

```diff
@@ -1412,6 +1412,15 @@
 behave as if they are implemented by calling \tcode{begin} and \tcode{end} on
 the \tcode{Range} and dispatching to the overload that takes separate
 iterator and sentinel arguments.
+
+\pnum
+\newtxt{Some algorithms declare both an overload that takes a \tcode{Range}
+and an \tcode{Iterator}, and an overload that takes two \tcode{Range}
+arguments. Since an array type~(\cxxref{basic.compound}) both satisifies
+\tcode{Range} and decays to a pointer~(\cxxref{conv.array}) which satisfies
+\tcode{Iterator}, such overloads are ambiguous when an array is passed as
+the second argument. Implementations shall provide a mechanism to resolve
+this ambiguity in favor of the overload that takes two ranges.}
```

Comment:
I think we also need a paragraph saying what part of the algorithm signatures can be relied upon and which cannot. Users are already prohibited from taking the address of `std` functions. We also need to say that the order of template parameters is unspecified, for instance. Whether an iterator or range is passed by value or by forwarding reference is technically observable -- the iterator's copy constructor might have side-effects -- so we need to be clear that they can't rely on iterators being copied into algorithms just because the signature that appears in the Ranges TS happens to give the impression that iterators will be taken by value.